### PR TITLE
Fixing ..\paths in XPlat

### LINKED ISSUE #913: Portable nuspec files
Ist there any way to use nuspec files with a platform dependend file-section on Windows and Mono/Unix?

Consider this nuspec file (https://github.com/fslaborg/FSharp.Charting):

        <?xml version="1.0" encoding="utf-8"?>											
	<package xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema">		
	  <metadata xmlns="http://schemas.microsoft.com/packaging/2010/07/nuspec.xsd">						
	    <id>@project@</id>													
	    <version>@build.number@</version>											
	    <authors>@authors@</authors>											
	    <owners>@authors@</owners>												
	    <licenseUrl>http://github.com/fslaborg/FSharp.Charting/blob/master/LICENSE.md</licenseUrl>				
	    <projectUrl>http://fslab.org/FSharp.Charting</projectUrl>								
	    <iconUrl>https://raw.github.com/fslaborg/FSharp.Charting/master/misc/logo.png</iconUrl>				
	    <requireLicenseAcceptance>false</requireLicenseAcceptance>								
	    <summary>@summary@</summary>											
	    <description>@description@</description>										
	    <releaseNotes>@releaseNotes@</releaseNotes>										
	    <copyright>Copyright 2013</copyright>										
	    <tags>@tags@</tags>													
	    <dependencies />													
	  </metadata>														
	  <files>														
	    <file src="..\bin\FSharp.Charting.Gtk.dll" target="lib/net40" />														
	    <file src="..\bin\FSharp.Charting.Gtk.xml" target="lib/net40" />							
	    <file src="..\bin\FSharp.Charting.Gtk.fsx" target="." />								
	    <file src="..\bin\OxyPlot.dll" target="lib/net40" />								
	    <file src="..\bin\OxyPlot.GtkSharp.dll" target="lib/net40" />							
	    <file src="..\bin\LICENSE.md" target="." />										
	  </files>														
	</package>														
	                                                                                                            
```NuGet.exe pack -Version 0.90.12 -OutputDirectory "/home/juergen/fsharp/FSharp.Charting/bin" FSharp.Charting.Gtk.0.90.12.nuspec```
Results in (even if you use ```MONO_IOMAP=all```):

      Attempting to build package from 'FSharp.Charting.Gtk.0.90.12.nuspec'.
      File not found: '..\bin\FSharp.Charting.Gtk.dll'.            

Replacing ```\``` with ```/```, still an error:

     Attempting to build package from 'FSharp.Charting.Gtk.0.90.12.nuspec'.
     Cannot create a package that has no dependencies nor content.

Adding ```-NoDefaultExcludes``` (why?) fixes this issue:

    NuGet.exe pack -NoDefaultExcludes  -Version 0.90.12 -OutputDirectory    "/home/juergen/fsharp/FSharp.Charting/bin" FSharp.Charting.Gtk.0.90.12.nuspec 
    Attempting to build package from 'FSharp.Charting.Gtk.0.90.12.nuspec'.
    Successfully created package    '/home/juergen/fsharp/FSharp.Charting/bin/FSharp.Charting.Gtk.0.90.12.nupkg'.

Any feedback is highly appreciated

### PR DESCRIPTION
**Pull request by toddm**
Fixes https://github.com/NuGet/Home/issues/913

This fixes XPlat pack so that it can find files referenced with ..\paths and also fixes the filter that removes files that start with a dot so that it doesn't match on files that start with ..\ or .\

I'm also moving PackCommand.cs to be with the other Command classes.

### CODE REVIEW

**Comment by toddm** [comment]
@emgarten @spadapet

File: src/NuGet.Core/NuGet.Common/PathResolver.cs [hunk 1]
```diff
@@ -57,6 +57,7 @@ private static Regex WildcardToRegex(string wildcard)
             {
                 // regex wildcard adjustments for *nix-style file systems
                 pattern = pattern
+                    .Replace(@"\.\*\*", @"\.[^/.]*") // .** should not match on ../file or ./file but will match .file
```
Comment (spadapet) [inline src/NuGet.Core/NuGet.Common/PathResolver.cs:60]: Shouldn't this apply to Windows too?

Comment (toddm) [inline src/NuGet.Core/NuGet.Common/PathResolver.cs:60]: I'm pretty sure that Windows is already working correctly but I'll reverify tomorrow.

**Comment by emgarten** [comment]
LGTM :shipit:

File: src/NuGet.Core/NuGet.Common/PathResolver.cs (hunk 1 above)
Comment (spadapet) [inline src/NuGet.Core/NuGet.Common/PathResolver.cs:60]: ok if Windows works then: :shipit:

**Pull request closed without merging**